### DISCUSSION TITLE
Choices does not support relate markdown

### DISCUSSION THREAD

**Discussion by KlavsKlavsen** [inducer/relate#930] [Q&A]
According to the docs, the Choices support relate markdown - which is documented here https://documen.tician.de/relate/content.html#markup - but if I try to give it this:
```
         choices:
           - ```python
             def f(x):
             return 5+x
             ```
```
it fails with:  Course content did not validate successfully: 'flows/skills-quiz-general.yml: ScannerError: while scanning for the next token found character '`' that cannot start any token in "", line 80, column 13: - ```python ^' Update not applied.

**Comment by KlavsKlavsen** [comment]
Even if I add such a section in the text for the question - it renders, but IMHO incorrectly as it looks like this rendered (also the case in the markdown test page):
```
A
<pre >  umbreallachart
    image
      tag v1.7.0
</pre> 
```
the pre html tags really should not be visible and the "block background color' - should form a box - like it does here on github, and not just cover the actual text inside the backticks.

**Reply by KlavsKlavsen** [reply]
the < pre > disappears IF I indent the backticks AND put a : after the line directly above (very odd)
```
          A:
 
             ```
             umbreallachart:
               image:
                 tag: v1.7.0
             ```
```

**Comment by inducer** [comment]
It does. But you need to use a YAML multi-line literal, like so:

```
type: ChoiceQuestion
id: fp_accuracy
shuffle: True
prompt: |
    # Floating point "machine epsilon"
    For a (binary) floating point system of the form
    $(s_1.s_2s_3)_2\cdot 2^{p}$ that has an exponent range from $-128$ to
    $127$ and that uses three bits to store the significand $s$, what is the
    difference between 1 and the smallest representable number greater than
    one?
choices:
    - $2^{-3}$
    - $2^{-4}$
    - |
      ```
      print("Hi")
      ```
    - ~CORRECT~  $2^{-2}$
answer_explanation: |

    That's just what it is.
```
Otherwise YAML will glue everything together into one line (without newlines) and Markdown will be mildly confused.

**Reply by KlavsKlavsen** [reply]
I tried to do that exact thing, and it complains about the backticks still

**Comment by KlavsKlavsen** [comment]
This fails to parse correctly too:
           ip a says:
 
 ```
             ```
             3: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
                link/ether 34:cf:f6:f5:8d:36 brd ff:ff:ff:ff:ff:ff
                inet 192.168.1.8/24 brd 192.168.1.255 scope global dynamic noprefixroute eth0
                valid_lft 1806101sec preferred_lft 1806101sec
             3: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
                link/ether 34:cf:f6:e5:8d:36 brd ff:ff:ff:ff:ff:ff
                inet 192.168.2.3/24 brd 192.168.2.255 scope global dynamic noprefixroute eth1
                valid_lft 1806101sec preferred_lft 1806101sec
                ```
 ```

**Reply by inducer** [reply]
Could you post a full question example? That'd make it easier for me to see what's going on. One thing to be aware of is that Markdown views "indentation by 4" as "code literal". (That's in the Markdown spec.) I saw some `<pre>`s in the output you posted, which got me suspicious.

**Comment by KlavsKlavsen** [comment]
full question:
```
     -
         type: ChoiceQuestion
         id: q2
 
         value: 4
         shuffle: true
         prompt: |
 
           # IP logic test 
           if I want to run curl http://dr.dk, on an ipv4 only box - which ips will we talk to, to perform this request.
 
           Resolv.conf has this:
 
             ```
             nameserver 10.55.55.10
             ```
 
           Route table looks like this:
 
             ```
             Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
             0.0.0.0         192.168.1.1     0.0.0.0         UG    600    0        0 eth0
             23.0.0.0        192.168.2.1     255.0.0.0       UG    600    0        0 eth1
             172.22.0.0      0.0.0.0         255.255.255.0   U     0      0        0 docker0
             192.168.1.0     0.0.0.0         255.255.255.0   U     600    0        0 eth0
             192.168.2.0     0.0.0.0         255.255.255.0   U     600    0        0 eth1
             ```
           'ip a' output:
 
             ```
             1: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
                link/ether 34:cf:f6:f5:8d:36 brd ff:ff:ff:ff:ff:ff
                inet 192.168.1.8/24 brd 192.168.1.255 scope global dynamic noprefixroute eth0
                valid_lft 1806101sec preferred_lft 1806101sec
             2: eth1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
                link/ether 34:cf:f6:e5:8d:36 brd ff:ff:ff:ff:ff:ff
                inet 192.168.2.3/24 brd 192.168.2.255 scope global dynamic noprefixroute eth1
                valid_lft 1806101sec preferred_lft 1806101sec
 
             ```
 
           arp -na says:
 
             ```
             arp -na
             ? (192.168.1.1) at ec:f0:fe:ea:92:40 [ether] on eth0
             ? (192.168.2.1) at ec:f0:fe:ea:91:40 [ether] on eth1
             ```
 
           host dr.dk says:
 
             ```
             dr.dk has address 23.38.201.196
             dr.dk has IPv6 address 2a02:26f0:c000:1a3::3f46
             dr.dk has IPv6 address 2a02:26f0:c000:1a4::3f46
             ```
 
         choices:
           - 10.55.55.10 and 23.38.201.196, both routed through 192.168.1.1
           - ~CORRECT~ 10.55.55.10, routed through 192.168.1.1 and 23.38.201.196,routed through 192.168.2.1
           - 23.38.201.196,routed through 192.168.1.1
           - 23.38.201.196 routed through 192.168.2.1
```

**Reply by inducer** [reply]
Not sure what you're trying to accomplish, but https://gist.github.com/inducer/155204d2506dc1a69a297e5e8bee0e78 looks reasonable to me.

**Comment by KlavsKlavsen** [comment]
![image](https://user-images.githubusercontent.com/2737885/161842590-84bd0047-f7ae-42bc-b4b1-7974fb0ce390.png)
This is how the 'ip a' output renders in my browser - using above markup :(

**Reply by inducer** [reply]
If I render the YAML from the gist in the sandbox, I get the following:

![grafik](https://user-images.githubusercontent.com/352067/161844412-ce1025c6-fdfa-480a-929f-42fb6a4a7aef.png)

**Reply by KlavsKlavsen** [reply]
in which browser?  I have tested this in both firefox and chrome and it renders the exact same way :(

**Reply by KlavsKlavsen** [reply]
[skills-quiz-general.zip](https://github.com/inducer/relate/files/8423698/skills-quiz-general.zip)

I have attached my full flows file so you can see.

**Reply by inducer** [reply]
It doesn't look like you're making use of my modifications from the gist.

**Comment by KlavsKlavsen** [comment]
![image](https://user-images.githubusercontent.com/2737885/162378670-bcbe05aa-3189-45fe-a532-4be11936810e.png)
I copy'pasted your gist in - and indented only (to match existing indention) - and it ALMOST loooks ok.
1) it shows <pre> tags
2) I do noget a big square around the 3xbackticked text like you have ?

**Reply by inducer** [reply]
Check if you have "may show untrusted HTML" set in the record for your course in the admin. I have a feeling this might be the HTML sanitization gone astray. (I'll take a look tomorrow.)

**Reply by KlavsKlavsen** [reply]
That did indeed fix it. Thank you for your help :)

**Reply by inducer** [reply]
#931 makes this work without that flag.
